src/source/Ice/IceAgent.h

```diff
@@ -123,22 +126,13 @@ typedef struct {
     UINT64 roundTripTime;
 } IceCandidatePair, *PIceCandidatePair;
 
-struct __TurnConnectionTracker {
-    struct __TurnConnection* pTurnConnection;
-
-    /* the ice candidate this TurnConnectionTracker is associated to. */
-    PIceCandidate pRelayCandidate;
-    /* store pointer to iceAgent to pass it to incomingDataHandler in incomingRelayedDataHandler
-     * we pass pTurnConnectionTrack as customData to incomingRelayedDataHandler to avoid look up
-     * turnConnectionTracker every time. */
-    PIceAgent pIceAgent;
-};
-
 struct __IceAgent {
     volatile ATOMIC_BOOL agentStartGathering;
     volatile ATOMIC_BOOL remoteCredentialReceived;
     volatile ATOMIC_BOOL candidateGatheringFinished;
     volatile ATOMIC_BOOL shutdown;
+    volatile ATOMIC_BOOL restart;
+    volatile ATOMIC_BOOL processStun;
```

Comment:
I think we can just use `restart` to signal the incoming data handler to not process stun packets. So, maybe we can remove `processStun`?

Reply:
user still need to be able to send and receive data using the existing connection while ice is restarting, so restart is not done until the next time a connected pair is found. whereas, we only need to block stun packets during the period in iceAgentRestart where it cleans up the local candidate and candidate pairs.

Reply:
Ah, I see, that makes sense.